src/containers/EffectList/EffectList.js

```diff
@@ -25,16 +26,49 @@ const cssMatchAction = css`
   margin-top: -1px;
 `
 
+const effectTypes = ['take', 'put', 'fork', 'call', 'cps', 'join', 'cancel']
+
 class EffectList extends React.Component {
 
   state = {
-    collapsedEffects: {}
+    collapsedEffects: {},
+    filter: { word: 'a', type: undefined }
   }
 
   isCollapsed = effectId => {
     return !!this.state.collapsedEffects[effectId]
   }
 
+  isFiltered = effectId => {
+    const { filter } = this.state
+
+    if (!filter.word) return true
+
+    let data = {}, description
+    const effect = this.props.effectsById[effectId]
+
+    if (effect.root) {
+      data = { type: 'root', name: effect.effect.saga.name }
+    }
+
+    else {
+      effectTypes.forEach(type => {
+        if (description = asEffect[type](effect.effect)) {
+          data = {
+            type,
+            name: description.pattern
+              || description.channel
+              || description.action
```

Comment:
sould be `description.action.name` ?

Reply:
sorry `description.action.type`

Reply:
Thanks for pointing it out. But, I wasn't sure with what are needed to be checked by the filter, because some of them have payload. So for now, I just made it as simple as possible. Do you think the filter better check for the payload including the keys and values as well?